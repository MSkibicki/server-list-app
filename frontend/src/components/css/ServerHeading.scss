@import "./Global.scss";

.server-heading-container {
  display: flex;
  align-items: center;
  width: 60%;
  justify-content: space-between;
  margin-top: 1em;

  .server-number {
    color: $nav-color;
    flex-direction: column;

    h1 {
      font-size: $server-heading-font;
    }

    h3 {
      font-size: $server-number-font;
    }
  }

  .server-input {
    margin-top: 2em;

    input {
      border: 2px $input-border-color solid;
      border-radius: 15px;
      background-color: $body-color;
      padding-left: 2em;
    }
  }
}
