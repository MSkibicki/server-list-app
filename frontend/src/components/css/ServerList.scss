@import "./Global.scss";

.server-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: $body-color;

  .server-list {
    width: 60%;
    padding: 0;
    margin: 0;
    color: $nav-color;
    min-height: 100vh;
  }
}
