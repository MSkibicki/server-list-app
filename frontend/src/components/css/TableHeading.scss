@import "./Global.scss";

.table-heading {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  width: 59.8%;
  background-color: $white;
  border: 2px solid $list-border-color;

  .name {
    width: 40%;
    color: $table-heading-color;
    font-weight: 600;
    text-transform: uppercase;
    font-size: $nav-font;
    margin-left: 3em;
  }

  .status {
    color: $table-heading-color;
    font-weight: 600;
    text-transform: uppercase;
    font-size: $nav-font;
  }
}

@media (max-width: 500px) {
  .table-heading {
    width: 79.6%;
  }
}
