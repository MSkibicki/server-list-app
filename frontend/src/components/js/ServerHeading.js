import React from "react";
import "../css/ServerHeading.scss";

const ServerHeading = ({ length, handleInput }) => {
  return (
    <div className="server-heading-container">
      <div className="server-number">
        <h1>Servers</h1>
        <h3>Number of elements: {length}</h3>
      </div>
      <div className="server-input">
        <input
          type="text"
          name="search"
          id="filter"
          placeholder="Search"
          onChange={handleInput}
          className="search"
        />
      </div>
    </div>
  );
};

export default ServerHeading;
