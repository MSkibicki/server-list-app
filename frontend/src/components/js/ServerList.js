import React, { useEffect, useState } from "react";
import "../css/ServerList.scss";
import axios from "axios";
import Loading from "./Loading";
import Server from "./Server";
import TableHeading from "./TableHeading";
import ServerHeading from "./ServerHeading";

const ServerList = () => {
  const [servers, setServers] = useState([]);
  const [loading, setLoading] = useState(false);
  const [inputValue, setInputValue] = useState("");

  useEffect(() => {
    let isCancelled = false;
    const getData = async () => {
      setLoading(true);

      try {
        const result = await axios.get("http://localhost:5000/servers");

        if (!isCancelled) {
          setLoading(false);
          setServers(result.data);
        }
      } catch (e) {
        if (!isCancelled) {
          console.error(e);
        }
      }
    };

    getData();

    return () => {
      isCancelled = true;
    };
  }, []);

  const handleInput = (e) => {
    setInputValue(e.target.value);
  };

  let filterServers = servers.filter((server) => {
    return server.name.toLowerCase().includes(inputValue.toLowerCase());
  });

  if (loading) return <Loading />;
  return (
    <div className="server-container">
      <ServerHeading length={servers.length} handleInput={handleInput} />
      <TableHeading />
      <ul className="server-list">
        {filterServers.map((filteredServer) => (
          <Server
            key={filteredServer.id}
            name={filteredServer.name}
            status={filteredServer.status}
          />
        ))}
      </ul>
    </div>
  );
};

export default ServerList;
