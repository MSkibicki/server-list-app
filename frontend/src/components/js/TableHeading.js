import React from "react";
import "../css/TableHeading.scss";

const TableHeading = () => {
  return (
    <div className="table-heading">
      <p className="name">name</p>
      <p className="status">status</p>
    </div>
  );
};

export default TableHeading;
