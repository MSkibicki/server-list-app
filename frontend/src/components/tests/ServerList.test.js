import ServerList from "./../js/ServerList";
import ServerHeading from "./../js/ServerHeading";
import TableHeading from "./../js/TableHeading";

describe("ServerList />", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<ServerList />);

    expect(wrapper).toMatchSnapshot();
  });

  it("has correct div classname", () => {
    const wrapper = shallow(<ServerList />);

    const divClass = wrapper
      .find("div.server-container")
      .hasClass("server-container");

    expect(divClass).toBe(true);
  });

  it("has correct ul classname", () => {
    const wrapper = shallow(<ServerList />);

    const ulClass = wrapper.find("ul.server-list").hasClass("server-list");

    expect(ulClass).toBe(true);
  });

  it("renders <ServerHeading /> component", () => {
    const wrapper = shallow(<ServerList />);

    const heading = wrapper.find(ServerHeading);

    expect(heading.length).toEqual(1);
  });

  it("renders <TableHeading /> component", () => {
    const wrapper = shallow(<ServerList />);

    const table = wrapper.find(TableHeading);

    expect(table.length).toEqual(1);
  });
});
