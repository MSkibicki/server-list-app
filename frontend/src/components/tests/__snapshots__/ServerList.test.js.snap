// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerList /> renders correctly 1`] = `
<div
  className="server-container"
>
  <ServerHeading
    handleInput={[Function]}
    length={0}
  />
  <TableHeading />
  <ul
    className="server-list"
  />
</div>
`;
